File: scripts/seed_profiles.py
import random
from datetime import datetime

import firebase_admin
from firebase_admin import credentials, firestore

# 🔑 Service Account
cred = credentials.Certificate("firebase-service-account.json")
firebase_admin.initialize_app(cred)

db = firestore.client()

REGIONS = ["LU", "ZH", "BE", "BS", "SG", "AG", "TG", "GR", "VS", "TI"]
ROLES = ["Singer", "Gitarre", "Bass", "Drums", "Keys", "DJ", "Violin"]
STATUSES = ["Band", "Solo", "Suchend"]
GENRES = ["Rock", "Metal", "Pop", "Indie", "Blues", "Jazz"]

def pick(arr):
    return random.choice(arr)

def pick_many(arr, min_n=1, max_n=3):
    return random.sample(arr, random.randint(min_n, min(max_n, len(arr))))

def seed_profiles(count=40):
    print(f"🌱 Seeding {count} test profiles…")

    for i in range(count):
        uid = f"test_{i+1}"
        name = f"TestMusiker {i+1}"
        band = f"Band {i+1}" if random.random() > 0.5 else ""

        avatar_id = (i % 70) + 1
        photo_url = f"https://i.pravatar.cc/300?img={avatar_id}"

        profile = {
            "uid": uid,
            "displayName": name,
            "photoURL": photo_url,
            "region": pick(REGIONS),
            "zip": str(6000 + random.randint(0, 300)),
            "bandName": band,
            "status": pick(STATUSES),
            "roles": pick_many(ROLES),
            "genres": pick_many(GENRES),
            "bio": "Dies ist ein automatisch generiertes Testprofil für Bandly.",
            "search": {
                "name": name.lower(),
                "band": band.lower(),
            },
            "isTest": True,
            "updatedAt": datetime.utcnow(),
        }

        db.collection("profiles").document(uid).set(profile)
        print(f"✅ {uid} · {name}")

    print("🎉 Fertig!")

if __name__ == "__main__":
    seed_profiles(40)

File: scripts/seed-profiles.js
/**
 * Run:
 *   node scripts/seed-profiles.cjs
 *
 * Needs .env.local with:
 *   NEXT_PUBLIC_FIREBASE_API_KEY
 *   NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN
 *   NEXT_PUBLIC_FIREBASE_PROJECT_ID
 */

require("dotenv").config({ path: ".env.local" });

const { initializeApp } = require("firebase/app");
const { getFirestore, doc, setDoc } = require("firebase/firestore");

// Minimal config reicht fürs Seeding
const firebaseConfig = {
  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
};

if (!firebaseConfig.apiKey || !firebaseConfig.authDomain || !firebaseConfig.projectId) {
  console.error("❌ Missing Firebase env vars. Check .env.local:");
  console.error("   NEXT_PUBLIC_FIREBASE_API_KEY, NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN, NEXT_PUBLIC_FIREBASE_PROJECT_ID");
  process.exit(1);
}

const app = initializeApp(firebaseConfig);
const db = getFirestore(app);

const REGIONS = ["LU", "ZH", "BE", "BS", "SG", "AG", "TG", "GR", "VS", "TI"];
const ROLES = ["Singer", "Gitarre", "Bass", "Drums", "Keys", "DJ", "Violin"];
const STATUSES = ["Band", "Solo", "Suchend"];
const GENRES = ["Rock", "Metal", "Pop", "Indie", "Blues", "Jazz"];

const pick = (arr) => arr[Math.floor(Math.random() * arr.length)];
const pickMany = (arr, min = 1, max = 3) => {
  const count = Math.floor(Math.random() * (max - min + 1)) + min;
  return [...arr].sort(() => 0.5 - Math.random()).slice(0, count);
};

// Entfernt undefined rekursiv (Firestore mag kein undefined)
function stripUndefined(obj) {
  if (Array.isArray(obj)) return obj.map(stripUndefined);
  if (obj && typeof obj === "object") {
    const out = {};
    for (const [k, v] of Object.entries(obj)) {
      if (v === undefined) continue;
      out[k] = stripUndefined(v);
    }
    return out;
  }
  return obj;
}

async function seed(count = 40) {
  console.log(`🌱 Seeding ${count} test profiles…`);

  for (let i = 0; i < count; i++) {
    const uid = `test_${i + 1}`;
    const name = `TestMusiker ${i + 1}`;
    const band = Math.random() > 0.5 ? `Band ${i + 1}` : "";

    const avatarId = (i % 70) + 1;
    const photoURL = `https://i.pravatar.cc/300?img=${avatarId}`;

    const profile = stripUndefined({
      uid,
      displayName: name,
      photoURL,
      region: pick(REGIONS),
      zip: String(6000 + Math.floor(Math.random() * 300)),
      bandName: band,
      status: pick(STATUSES),
      roles: pickMany(ROLES),
      genres: pickMany(GENRES),
      bio: "Dies ist ein automatisch generiertes Testprofil für Bandly.",
      search: {
        name: name.toLowerCase(),
        band: band.toLowerCase(),
      },
      isTest: true,
      // ✅ statt serverTimestamp() -> Date ist Firestore-kompatibel
      updatedAt: new Date(),
    });

    try {
      await setDoc(doc(db, "profiles", uid), profile, { merge: true });
      console.log(`✅ ${uid} · ${name}`);
    } catch (err) {
      console.error(`❌ Failed on ${uid}`, err);
      process.exit(1);
    }
  }

  console.log("🎉 Fertig!");
}

seed(40);
